orca-front50/src/main/groovy/com/netflix/spinnaker/orca/front50/tasks/MonitorFront50Task.java

```diff
@@ -88,9 +94,14 @@ public TaskResult execute(@Nonnull Stage stage) {
           }
 
           Long lastModifiedTime = Long.valueOf(pipeline.get().get("updateTs").toString());
-          if (lastModifiedTime < stage.getStartTime()) {
+          if (lastModifiedTime < (stage.getStartTime() - gracePeriodMs)) {
             return TaskResult.RUNNING;
           }
+
+          try {
+            // small delay between verification attempts
+            Thread.sleep(1000);
```

Comment:
It retrieves the pipeline from front50 multiple times (assume it will hit various front 50 instances).  If it gets the updated pipeline 5 times in a row (with a second in between each fetch), then we're happy.   If it gets a stale copy even once, reschedule the whole the stage.

That right?

Reply:
Yar.